actors/builtin/miner/miner_test.go

```diff
@@ -1703,6 +1706,67 @@ func TestTerminateSectors(t *testing.T) {
 			assert.Equal(t, big.Zero(), st.InitialPledgeRequirement)
 		}
 	})
+
+	t.Run("charges correct fee for young termination of committed capacity upgrade", func(t *testing.T) {
+		actor := newHarness(t, periodOffset)
+		rt := builderForHarness(actor).
+			WithBalance(bigBalance, big.Zero()).
+			Build(t)
+		actor.constructAndVerify(rt)
+
+		// A miner will pay the minimum of termination fee and locked funds. Add some locked funds to ensure
+		// correct fee calculation is used.
+		actor.addLockedFunds(rt, big.Mul(big.NewInt(1e18), big.NewInt(20000)))
+
+		// Move the current epoch forward so that the first deadline is a stable candidate for both sectors
+		rt.SetEpoch(periodOffset + miner.WPoStChallengeWindow)
+
+		// Commit a sector to upgrade
+		// Use the max sector number to make sure everything works.
+		oldSector := actor.commitAndProveSector(rt, abi.MaxSectorNumber, defaultSectorExpiration, nil)
+		st := getState(rt)
+		dlIdx, partIdx, err := st.FindSector(rt.AdtStore(), oldSector.SectorNumber)
+		require.NoError(t, err)
+
+		// Reduce the epoch reward so that a new sector's initial pledge would otherwise be lesser.
+		actor.epochReward = big.Div(actor.epochReward, big.NewInt(2))
+		actor.epochRewardSmooth = smoothing.TestingConstantEstimate(actor.epochReward)
+
+		challengeEpoch := rt.Epoch() - 1
+		upgradeParams := actor.makePreCommit(200, challengeEpoch, oldSector.Expiration, []abi.DealID{1})
+		upgradeParams.ReplaceCapacity = true
+		upgradeParams.ReplaceSectorDeadline = dlIdx
+		upgradeParams.ReplaceSectorPartition = partIdx
+		upgradeParams.ReplaceSectorNumber = oldSector.SectorNumber
+		upgrade := actor.preCommitSector(rt, upgradeParams)
+
+		// Prove new sector
+		rt.SetEpoch(upgrade.PreCommitEpoch + miner.PreCommitChallengeDelay + 1)
+		newSector := actor.proveCommitSectorAndConfirm(rt, &upgrade.Info, upgrade.PreCommitEpoch,
+			makeProveCommit(upgrade.Info.SectorNumber), proveCommitConf{})
+
+		// Expect replace parameters have been set
+		assert.Equal(t, oldSector.ExpectedDayReward, newSector.ReplacedDayReward)
+		assert.Equal(t, rt.Epoch()-oldSector.Activation, newSector.ReplacedSectorAge)
+
+		// advance clock a little and terminate new sector
+		rt.SetEpoch(rt.Epoch() + 100)
+		sectorPower := miner.QAPowerForSector(actor.sectorSize, newSector)
+		twentyDayReward := miner.ExpectedRewardForPower(actor.epochRewardSmooth, actor.epochQAPowerSmooth, sectorPower, miner.InitialPledgeProjectionPeriod)
+		sectorAge := rt.Epoch() - newSector.Activation
+		expectedFee := miner.PledgePenaltyForTermination(
+			newSector.ExpectedDayReward,
+			oldSector.ExpectedDayReward,
+			twentyDayReward,
+			sectorAge,
+			newSector.Activation-oldSector.Activation,
```

Comment:
Please break this out as `oldSectorAge`, it looks wrong otherwise.

From my note above, if we use the correct expiration epoch for the old sector, it will be independent of the new sector's activation.